src/main/java/de/komoot/photon/PhotonDoc.java

```diff
@@ -116,4 +126,60 @@ public boolean isUsefulForIndex() {
 
         return true;
     }
+    
+    /**
+     * Complete doc from nominatim address information.
+     */
+    public void completeFromAddress() {
+        String addressStreet = address != null ? address.get("street") : null;
+        if (addressStreet != null) {
+            if (this.street == null) {
+                this.street = new HashMap<>();
+            }
+            setOrReplace(addressStreet, this.street, "street");
+        }
+        
+        String addressCity = address != null ? address.get("city") : null;
+        if (addressCity != null) {
+            if (this.city == null) {
+                this.city = new HashMap<>();
+            }
+            setOrReplace(addressCity, this.city, "city");
+        }
+        
+        String addressSuburb = address != null ? address.get("suburb") : null;
+        if (addressSuburb != null) {
+            if (this.suburb == null) {
+                this.suburb = new HashMap<>();
+            }
+            setOrReplace(addressSuburb, this.suburb, "suburb");
+        }
+        
+        String addressNeighbourhood = address != null ? address.get("neighbourhood") : null;
+        if (addressNeighbourhood != null) {
+            if (this.neighbourhood == null) {
+                this.neighbourhood = new HashMap<>();
+            }
+            setOrReplace(addressNeighbourhood, this.neighbourhood, "neighbourhood");
+        }
```

Comment:
This code duplication can be avoided if you move the lookup into setOrReplace as well.

Reply:
I've removed _some_ duplication but not as much as I would have liked as the assignment to the instance fields (`this.locality`...) must happen outside `setOrReplace` or am I overlooking something?

Reply:
I was thinking along the lines of having a function which can be used like this:

    this.locality = extractAddress(this.locality, address, "neighbourhood")

Nevermind for now. It is good as is.